### PR TITLE
unopened isolated positions pt1

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
Figma Design: https://www.figma.com/design/mKevZOfE9nj6MZpiolKYW1/dYdX-%E2%80%BA-Mobile?node-id=5962-23534&t=amLuAKC8dnKISGXT-4

<br/>

## Description / Intuition
part 2 will be the cancel behavior and the market info appearance ([see comment](https://www.figma.com/design/mKevZOfE9nj6MZpiolKYW1?node-id=5962-28819#839405365))
- wires up `pendingPositions` to display in Overview and Positions lists
- refactors `positions` to not be a `PlatformListView` since there are two `items` list
- renames `ParsingError` computed property to `localizedMessage` to avoid confusion with system `localizedDescription` API
- adds local validation to [match web](https://github.com/dydxprotocol/v4-web/blob/6d1e9bf71ba0c7010edd801165ecc8d5d9928072/src/views/forms/AdjustIsolatedMarginForm.tsx#L178) since it is not in abacus yet
- removes "[Max](https://dydx-team.slack.com/archives/C06ETJJGYLX/p1718386918927559?thread_ts=1718386847.629689&cid=C06ETJJGYLX)" button

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
| ensuring feature flag still works |  |
| ensuring feature flag still works |  |
| display unopened isolated positions in positions |  |
| display unopened isolated positions in overview |  |

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxAdjustMarginInputViewBuilder.swift [hunk 1]
```diff
@@ -225,14 +270,6 @@ private class dydxAdjustMarginInputViewPresenter: HostedViewPresenter<dydxAdjust
...
     }
 
-    private func updateButtonState(input: AdjustIsolatedMarginInput) {
-        if parser.asNumber(input.amount)?.doubleValue ?? 0 > 0 {
-            self.ctaButtonPresenter.viewModel?.ctaButtonState = .enabled()
-        } else {
-            self.ctaButtonPresenter.viewModel?.ctaButtonState = .disabled()
-        }
-    }
-
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxAdjustMarginInputViewBuilder.swift:235]: this was moved to validate fxn

File: dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxAdjustMarginInputViewBuilder.swift [hunk 2]
```diff
@@ -80,16 +80,13 @@ private class dydxAdjustMarginInputViewPresenter: HostedViewPresenter<dydxAdjust
         viewModel.amount?.onEdited = { amount in
             AbacusStateManager.shared.adjustIsolatedMargin(input: amount, type: .amount)
         }
-        viewModel.amount?.maxAction = {
-            AbacusStateManager.shared.adjustIsolatedMargin(input: "1", type: .amountpercent)
-        }
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxAdjustMarginInputViewBuilder.swift:85]: per corey's [guidance](https://dydx-team.slack.com/archives/C06ETJJGYLX/p1718386918927559?thread_ts=1718386847.629689&cid=C06ETJJGYLX), removing max button

File: dydx/dydxViews/dydxViews/_v4/Portfolio/Components/Sections/dydxPortfolioPendingPositionsItemViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        }
+    }
+
+    public override func createView(parentStyle: ThemeStyle = ThemeStyle.defaultStyle, styleKey: String? = nil) -> PlatformView {
+        PlatformView(viewModel: self, parentStyle: parentStyle, styleKey: styleKey) { [weak self] _  in
+            guard let self = self else { return AnyView(PlatformView.nilView) }
+
+            return VStack(spacing: 0) {
+                self.topContent
+                self.divider
```
Comment (ruixhuang) [inline dydx/dydxViews/dydxViews/_v4/Portfolio/Components/Sections/dydxPortfolioPendingPositionsItemViewModel.swift:109]: Doesn't seem like the divider is visible from the screenshot

**Review by ruixhuang — DISMISSED** [review]

File: dydx/dydxViews/dydxViews/_v4/Portfolio/Components/Sections/dydxPortfolioPendingPositionsItemViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    private var divider: some View {
+        Spacer(minLength: 1)
+            .overlay(ThemeColor.SemanticColor.borderDefault.color)
+    }
+
+    private var bottomContent: some View {
+        let viewOrdersStringKey: String
+        let viewOrdersStringParams: [String: String]?
+        if orderCount > 1 {
```
Comment (ruixhuang) [inline dydx/dydxViews/dydxViews/_v4/Portfolio/Components/Sections/dydxPortfolioPendingPositionsItemViewModel.swift]: orderCount == 1

File: dydx/dydxViews/dydxViews/_v4/Portfolio/Components/Sections/dydxPortfolioPendingPositionsItemViewModel.swift (hunk 3 above)
Comment (mike-dydx) [reply dydx/dydxViews/dydxViews/_v4/Portfolio/Components/Sections/dydxPortfolioPendingPositionsItemViewModel.swift:109]: addressed in de39f47380e9b165ff43b1afa2156f96029a9dd4
Screenshot 2024-06-17 at 1 56 50 PM

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**